hoverxref/extension.py

```diff
@@ -174,7 +174,7 @@ def setup_intersphinx(app, config):
 
     for listener in app.events.listeners.get('missing-reference'):
         module_name = inspect.getmodule(listener.handler).__name__
-        if module_name == 'sphinx.ext.intersphinx':
+        if module_name.startswith('sphinx.ext.intersphinx'):
```

Comment:
I found the `module_name` is called `sphinx.ext.intersphinx._resolve`. Would it be better to be pretty explicit here to avoid disconnecting things we are not expecting to disconnect?

```suggestion
        if module_name == 'sphinx.ext.intersphinx._resolve':
```

Reply:
This explicit name check would be compatible only with Sphinx-7.4+. 

We could check explicitly for `_resolve` first, then the old name as a fallback for backwards compatibility, or according to the Sphinx version. What do you think? 

Initially I was moving towards not having a private module in the name check but what you highlight regarding risk of other listeners being disconnected is true, so happy to make the update!

Reply:
Good point! Yeah, I think it's better to check for the Sphinx version and use one name or the other depending on that 👍🏼